src/go/rpk/pkg/cli/redpanda/admin/brokers/decommission-status.go

```diff
@@ -20,7 +22,47 @@ func newDecommissionBrokerStatus(fs afero.Fs, p *config.Params) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "decommission-status [BROKER ID]",
 		Short: "Show the progress of a node decommissioning",
-		Args:  cobra.ExactArgs(1),
+		Long: `Show the progrss of a node decommissioning.
+
+When a node is being decommissioned, this command reports the decommissioning
+progress as follows.
+
+$ rpk redpanda admin brokers decommission-status 4
+DECOMMISSION PROGRESS
+=====================
+NAMESPACE-TOPIC              PARTITION  MOVING-TO  COMPLETION-%  PARTITION-SIZE
+kafka/test                   0          3          9             1699470920
+kafka/test                   4          3          0             1614258779
+kafka/test2                  3          3          3             2722706514
+kafka/test2                  4          3          4             2945518089
+kafka_internal/id_allocator  0          3          0             3562
+
+Using --detailed / -d, it prints granular reports additionally.
+
+$ rpk redpanda admin brokers decommission-status 4 -d
+DECOMMISSION PROGRESS
+=====================
+NAMESPACE-TOPIC  PARTITION  MOVING-TO  COMPLETION-%  PARTITION-SIZE  BYTES-MOVED  BYTES-REMAINING
+kafka/test       0          3          13            1731773243      228114727    1503658516
+kafka/test       4          3          1             1645952961      18752660     1627200301
+kafka/test2      3          3          5             2752632301      140975805    2611656496
+kafka/test2      4          3          6             2975443783      181581219    2793862564
+
+If a partition cannot be moved with some reason, the command reports the
```

Comment:
Any way to get these reasons?

Reply:
It typically occurs when something wrong happens in raft. Report it to the support team for further assistance?
@bharathv , wdyt? Should we encourage the user to grab the /v1/debug/partition output and report to us?

Reply:
Just read the core PR, https://github.com/redpanda-data/redpanda/pull/11899.

> A very common example is decommissioning a node from 3 node cluster with rf=3 partitions. In such cases the API does not report anything but decommissioning is hung.

Should we just mention this scenario?

Reply:
The reason is typically outside of raft, like no node with enough disk space or no broker that can satisfy rack constraint etc. Currently there is no easy way to get the exact reason.  "allocation" failure in this context means there is no suitable broker to allocate a replica of this partition. Perhaps we could rephrase it to something more meaningful to a user. Open to suggestions.

Reply:
Oh, thanks for the correction. Then is it fair to rephrase in the output to something like this? I know it's a bit long though.

> ALLOCATION FAILURE

to 

> NO BROKER TO ALLOCATE

Reply:
I think allocation failure is fine, so long as it's somewhat slightly explained